tools/dds/dds-adapter/lrs-device-controller.cpp

```diff
@@ -491,48 +505,93 @@ lrs_device_controller::lrs_device_controller( rs2::device dev, std::shared_ptr<
     // Create a supported streams list for initializing the relevant DDS topics
     auto supported_streams = get_supported_streams();
 
-    _bridge.on_start_sensor( [this]( std::string const & sensor_name, dds_stream_profiles const & active_profiles ) {
-        auto & sensor = _rs_sensors[sensor_name];
-        auto rs2_profiles = get_rs2_profiles( active_profiles );
-        sensor.open( rs2_profiles );
-        sensor.start( [this]( rs2::frame f ) {
-            auto stream_name = stream_name_from_rs2( f.get_profile() );
-            auto it = _stream_name_to_server.find( stream_name );
-            if( it != _stream_name_to_server.end() )
+    _bridge.on_start_sensor(
+        [this]( std::string const & sensor_name, dds_stream_profiles const & active_profiles )
+        {
+            auto & sensor = _rs_sensors[sensor_name];
+            auto rs2_profiles = get_rs2_profiles( active_profiles );
+            sensor.open( rs2_profiles );
+            if( sensor.is< rs2::motion_sensor >() )
             {
-                auto & server = it->second;
-                if( _bridge.is_streaming( server ) )
+                struct imu_context
                 {
-                    realdds::topics::image_msg image;
-                    auto data = static_cast< const uint8_t * >( f.get_data() );
-                    image.raw_data.assign( data, data + f.get_data_size() );
-                    image.height = server->get_image_header().height;
-                    image.width = server->get_image_header().width;
-                    image.timestamp  // in sec.nsec
-                        = realdds::dds_time( static_cast< long double >( f.get_timestamp() ) / 1e3 );
-                    if( auto video = std::dynamic_pointer_cast< realdds::dds_video_stream_server >( server ) )
+                    realdds::topics::imu_msg message;
+                    // We will be getting gyro and accel together, from different threads:
+                    // Don't want to change something mid-send
+                    std::mutex mutex;
+                };
+                sensor.start(
+                    [this, imu = std::make_shared< imu_context >()]( rs2::frame f )
+                    {
+                        rs2::stream_profile stream_profile;
+                        auto motion = std::dynamic_pointer_cast< realdds::dds_motion_stream_server >(
+                            frame_to_streaming_server( f, &stream_profile ) );
+                        if( ! motion )
+                            return;
+
+                        auto xyz = reinterpret_cast< float const * >( f.get_data() );
+                        if( RS2_STREAM_ACCEL == stream_profile.stream_type() )
+                        {
+                            std::unique_lock< std::mutex > lock( imu->mutex );
```

Comment:
I think that `std::lock_guard` is more efficient

Reply:
Umm, yes, I don't know how it got there. But essentially the same, maybe slightly more overhead. I'll fix if other changes are needed.